src/material-date-fns-adapter/adapter/date-fns-adapter.ts

```diff
@@ -241,4 +246,42 @@ export class DateFnsAdapter extends DateAdapter<Date, Locale> {
   invalid(): Date {
     return new Date(NaN);
   }
+
+  override setTime(target: Date, hours: number, minutes: number, seconds: number): Date {
+    if (typeof ngDevMode === 'undefined' || ngDevMode) {
+      if (hours < 0 || hours > 23) {
+        throw Error(`Invalid hours "${hours}". Hours value must be between 0 and 23.`);
+      }
+
+      if (minutes < 0 || minutes > 59) {
+        throw Error(`Invalid minutes "${minutes}". Minutes value must be between 0 and 59.`);
+      }
+
+      if (seconds < 0 || seconds > 59) {
+        throw Error(`Invalid seconds "${seconds}". Seconds value must be between 0 and 59.`);
+      }
+    }
+
+    return set(this.clone(target), {hours, minutes, seconds});
+  }
+
+  override getHours(date: Date): number {
+    return getHours(date);
+  }
+
+  override getMinutes(date: Date): number {
+    return getMinutes(date);
+  }
+
+  override getSeconds(date: Date): number {
+    return getSeconds(date);
+  }
+
+  override parseTime(value: any, parseFormat: string | string[]): Date | null {
+    return this.parse(value, parseFormat);
+  }
+
+  override addMilliseconds(date: Date, amount: number): Date {
```

Comment:
should this be `addSeconds`? since it doesn't look like we really try to deal wit millisecond precision?

Reply:
I was thinking we might use the milliseconds for something else in the future, but we can deal with it if it comes up. Changed.